### PR TITLE
Add tests for filtering

### PR DESCRIPTION
**Pull request by dblenkus**

### CODE REVIEW

File: resolwe/flow/tests/test_filtering.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        pass  # XXX: `modified` is changed on save
+
+    def test_filter_data(self):
+        self._apply_filter({'data': self.data.pk}, [self.collection_1, self.collection_2])
+
+    def test_filter_descriptor_schema(self):
+        self._apply_filter({'descriptor_schema': self.descriptor_schema_1.pk}, [self.collection_1, self.collection_2])
+
+    def test_filter_description(self):
+        pass  # XXX: filtering by descriptor fields is not supported in django-filter package
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_filtering.py]: Do we need this test then?

File: resolwe/flow/tests/test_filtering.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        self._apply_filter({'contributor': self.user_1.pk},
+                           [self.collection_1, self.collection_2])
+
+    def test_filter_created(self):
+        self._apply_filter({'created__year': '2016'}, [self.collection_1, self.collection_2])
+        self._apply_filter({'created__year__gt': '2015'}, [self.collection_1, self.collection_2])
+        self._apply_filter({'created__year__gt': '2015', 'created__month__gte': '8'}, [self.collection_2])
+
+    def test_filter_modified(self):
+        pass  # XXX: `modified` is changed on save
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_filtering.py]: You can still count the number of objects that you have created until now and that you have not created anything from now on.

**Comment by mstajdohar** [comment]
Rebase onto master and accept.

**Comment by dblenkus** [comment]
Fixed comments.

**Pull request merged**